department-of-veterans-affairs/vets-website
# 74348b Add Medicare status page custom component

## Summary

This PR introduces a custom page component for use in form 10-10D, as well as a few small helper functions that will be used by this and upcoming custom components/form config changes. This component will be added to form 10-10D's Applicant Information section in an upcoming PR. For now, it is just being added without being integrated so that the PR stays within a reasonable size. 

The key contribution of this component is that it facilitates custom wording that reflects changes in tense and person based on form state. For example, it allows us to change the question wording to say "Do you have medicare" or "Does Jim have medicare" based on whether or not the application is being filled out by a third party. 

- Added a custom page for collecting Medicare information with dynamic wording
- Added a custom review page component
- Added a small helper function that will be used across future components to standardize wording
- **Team**: IVC-CHAMPVA
- **Flipper**: NA, form is not set to go to prod

## Related issue(s)

- [_Link to ticket created in va.gov-team repo_](https://github.com/department-of-veterans-affairs/va.gov-team/issues/74348)

## Testing done

Additional testing is planned when this component is integrated in the form. For the time being, the following has been done:
- Manual testing to verify fields behave as expected
- Ran `yarn test:coverage-app ivc-champva/10-10D` to verify existing tests run and pass
- Added unit test for helper function

## Screenshots

Example of wording functionality:

|         | Scenario 1 | Scenario 2 |
| ------- | ------ | ----- |
| Desktop | ![med0](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/03702f27-2be6-46a5-97ad-25c2dc4892e6) | ![med1](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/64d7cf7d-1c05-44fb-aab7-4615ffe18cee) |

## What areas of the site does it impact?

This form only

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

NA
